Client/InfluxDBClientOptions.cs

```diff
@@ -23,29 +23,211 @@ public class InfluxDBClientOptions
             RegexOptions.CultureInvariant |
             RegexOptions.RightToLeft);
 
-        public string Url { get; }
-        public LogLevel LogLevel { get; }
+        private char[] _token;
+        private string _url;
+        private TimeSpan _timeout;
+        private LogLevel _logLevel;
+        private string _username;
+        private char[] _password;
+        private IWebProxy _webProxy;
+        private bool _allowHttpRedirects;
+        private bool _verifySsl;
+        private X509CertificateCollection _clientCertificates;
 
-        public AuthenticationScheme AuthScheme { get; }
-        public char[] Token { get; }
-        public string Username { get; }
-        public char[] Password { get; }
+        /// <summary>
+        /// Set the url to connect the InfluxDB.
+        /// </summary>
+        public string Url
+        {
+            get => _url;
+            set
+            {
+                Arguments.CheckNonEmptyString(value, "Url");
+                _url = value;
+            }
+        }
 
-        public string Org { get; }
-        public string Bucket { get; }
+        /// <summary>
+        /// Set the timespan to wait before the HTTP request times out.
+        /// </summary>
+        public TimeSpan Timeout
+        {
+            get => _timeout;
+            set
+            {
+                Arguments.CheckNotNull(value, "Timeout");
+                _timeout = value;
+            }
+        }
 
-        public TimeSpan Timeout { get; }
+        /// <summary>
+        /// Set the log level for the request and response information.
+        /// </summary>
+        public LogLevel LogLevel
+        {
+            get => _logLevel;
+            set
+            {
+                Arguments.CheckNotNull(value, "LogLevel");
+                _logLevel = value;
+            }
+        }
 
-        public IWebProxy WebProxy { get; }
-        public bool AllowHttpRedirects { get; }
+        public AuthenticationScheme AuthScheme { get; private set; }
+
+        /// <summary>
+        /// Setup authorization by <see cref="AuthenticationScheme.Token"/>.
+        /// </summary>
+        public char[] Token
```

Comment:
The InfluxDBClientOptions should also accept `string` as a Token. This should be valid code:

```c#
var options = new InfluxDBClientOptions("http://localhost:8086")
{
    Token = "my-token",
    Org = "my-org"
};  
using var client = new InfluxDBClient(options);
```